x-pack/plugins/observability_solution/apm_data_access/server/services/get_host_names/index.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import { HOST_NAME } from '@kbn/apm-types/es_fields';
+import type { TimeRangeMetadata } from '../../../common';
+import { getBucketSize } from '../../../common/utils/get_bucket_size';
+import { getPreferredBucketSizeAndDataSource } from '../../../common/utils/get_preferred_bucket_size_and_data_source';
+import { ApmDocumentType } from '../../../common/document_type';
+import type { ApmDataAccessServicesParams } from '../get_services';
+
+const MAX_SIZE = 1000;
+
+export interface HostNamesRequest {
+  query: estypes.QueryDslQueryContainer;
+  kuery?: string;
+  start: number;
+  end: number;
+  size?: number;
+  documentSources: TimeRangeMetadata['sources'];
+}
+
+const suitableTypes = [ApmDocumentType.TransactionMetric];
+
+export function createGetHostNames({ apmEventClient }: ApmDataAccessServicesParams) {
+  return async ({ start, end, size = MAX_SIZE, query, documentSources }: HostNamesRequest) => {
+    const sourcesToUse = getPreferredBucketSizeAndDataSource({
+      sources: documentSources.filter((s) => suitableTypes.includes(s.documentType)),
+      bucketSizeInSeconds: getBucketSize({ start, end, numBuckets: 50 }).bucketSize,
```

Comment:
should this be `(end - start) * 1000`?

Reply:
makes sense. thanks

Reply:
I had to revert this change, with `(end - start) * 1000`, `getPreferredBucketSizeAndDataSource` was returning the wrong `rollupInterval` (10m instead of 1m) with 15m range